### PR TITLE
Featulre/chat skeleton

### PR DESCRIPTION
**Pull request by DavidSurkov**
image
Chat

### CODE REVIEW

File: src/pages/Chat/Chat.tsx [hunk 1]
```diff
@@ -0,0 +1,95 @@
+import { useState } from "react";
+import {
+  IChatUser,
+  useGetRoomsQuery,
+} from "services/notifications/setNotificationsAPI";
+import { IUser, useGetOwnUserQuery } from "services/user/setUserAPI";
+import { Block, Container, Notification } from "pages/Chat/Chat.styles";
+import { useTranslation } from "react-i18next";
+import ChatRoomsList from "./ChatRoomsList/ChatRoomsList";
+import ChatRoom from "./ChatRoom/ChatRoom";
```
Comment (xx-TOOL-xx) [inline src/pages/Chat/Chat.tsx]: imports

File: src/pages/Chat/Chat.tsx [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+    setRoomId(id);
+    setJobNameForHeader(jobName);
+  };
+
+  const defineOpponentsNameAndAvatar = (
+    freelancer: IChatUser,
+    client: IChatUser,
+    currentUser: IUser
+  ): { name: string; avatar: string } => {
+    if (+currentUser.id!! === freelancer.id) {
```
Comment (xx-TOOL-xx) [inline src/pages/Chat/Chat.tsx]: one exclamation mark is not enough for ts?

File: src/pages/Chat/Chat.tsx [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+                id={room.id}
+                onClick={onClickHandler}
+                jobName={room.jobPost.title}
+              />
+            ))}
+          </Block>
+          {roomId ? (
+            <ChatRoom
+              roomUsers={roomUsers}
+              userSelfId={+user.id!!}
```
Comment (xx-TOOL-xx) [inline src/pages/Chat/Chat.tsx]: same

File: src/services/notifications/setNotificationsAPI.ts [hunk 4]
```diff
@@ -1,7 +1,71 @@
...
+    isSystemMessage: boolean;
+    text: string;
+  };
+}
+
+interface ISendMessage {
+  chatRoomId: number;
+  text: string;
+}
+
```
Comment (xx-TOOL-xx) [inline src/services/notifications/setNotificationsAPI.ts]: it's 70 lines only types. can it be in separate from logic?

**Review by xx-TOOL-xx — APPROVED** [review]

File: src/pages/Chat/Chat.tsx (hunk 1 above)
Comment (DavidSurkov) [reply src/pages/Chat/Chat.tsx]: done

File: src/services/notifications/setNotificationsAPI.ts (hunk 4 above)
Comment (DavidSurkov) [reply src/services/notifications/setNotificationsAPI.ts]: done

File: src/pages/Chat/Chat.tsx (hunk 2 above)
Comment (DavidSurkov) [reply src/pages/Chat/Chat.tsx]: +

File: src/pages/Chat/Chat.tsx (hunk 3 above)
Comment (DavidSurkov) [reply src/pages/Chat/Chat.tsx]: +

**Comment by DavidSurkov** [comment]
image

File: src/services/notifications/setNotificationsAPI.ts [hunk 5]
```diff
@@ -47,13 +48,113 @@ const notificationsAPI = apiSlice.injectEndpoints({
...
+          });
+
+          await cacheEntryRemoved;
+
+          socket.off(EventEnum.ROOMS);
+        } catch {
+          throw new Error("Error, cannot get chat rooms");
+        }
+      },
+    }),
```
Comment (lyudad) [inline src/services/notifications/setNotificationsAPI.ts:80]: look like airbnb eslint  rules was not applied

**Pull request merged**